msticpy/data/drivers/sumologic_driver.py

```diff
@@ -0,0 +1,481 @@
+}
+
+
+@export
+class SumologicDriver(DriverBase):
+    """Driver to connect and query from Sumologic."""
+
+    _SUMOLOGIC_REQD_ARGS = ["url", "accessid", "accesskey"]
+    _CONNECT_DEFAULTS: Dict[str, Any] = {"url": "https://api.us2.sumologic.com/api"}
+    _TIME_FORMAT = '"%Y-%m-%d %H:%M:%S.%6N"'
+    timeout = 300
+
+    def __init__(self, **kwargs):
+        """Instantiate Sumologic Driver."""
+        super().__init__()
+        self.service = None
+        self._loaded = True
+        self._connected = False
+        self._debug = kwargs.get("debug", False)
+        self.public_attribs = {
+            "client": self.service,
+            "saved_searches": self._saved_searches,
+            "fired_alerts": self._fired_alerts,
+        }
+        self.formatters = {"datetime": self._format_datetime, "list": self._format_list}
+
+    def connect(self, connection_str: str = None, **kwargs):
+        """
+        Connect to Sumologic via sumologic-sdk.
+
+        Parameters
+        ----------
+        connection_str : Optional[str], optional
+            Connection string with Sumologic connection parameters
+
+        Other Parameters
+        ----------------
+        kwargs :
+            Connection parameters can be supplied as keyword parameters.
+
+        Notes
+        -----
+        Default configuration is read from the DataProviders/Sumologic
+        section of msticpyconfig.yaml, if available.
+
+        """
+        cs_dict = self._get_connect_args(connection_str, **kwargs)
+
+        arg_dict = {
+            key: val for key, val in cs_dict.items() if key in SUMOLOGIC_CONNECT_ARGS
+        }
+        try:
+            # https://github.com/SumoLogic/sumologic-python-sdk/blob/master/scripts/search-job.py
+            self.service = SumoLogic(
+                arg_dict['accessid'],
+                arg_dict['accesskey'],
+                arg_dict['url']
+            )
+        except AuthenticationError as err:
+            raise MsticpyConnectionError(
+                f"Authentication error connecting to Sumologic: {err}",
+                title="Sumologic connection",
+                help_uri="https://msticpy.readthedocs.io/en/latest/DataProviders.html",
+            )
+        except HTTPError as err:
+            raise MsticpyConnectionError(
+                f"Communication error connecting to Sumologic: {err}",
+                title="Sumologic connection",
+                help_uri="https://msticpy.readthedocs.io/en/latest/DataProviders.html",
+            )
+        except Exception as err:
+            raise MsticpyConnectionError(
+                f"Error connecting to Sumologic: {err}",
+                title="Sumologic connection",
+                help_uri="https://msticpy.readthedocs.io/en/latest/DataProviders.html",
+            )
+        self._connected = True
+        print("connected")
+
+    def _get_connect_args(
+        self, connection_str: Optional[str], **kwargs
```

Comment:
I can't see where connection_str is used in this function?

Reply:
likely inherited from other driver template and to be removed